### PR TITLE
Add functional test for UI Metric

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

Add functional tests to validate the UI Metric Collector introduced in OpenSearch dashboards core.

**Tests**

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/23f24815-0084-4448-bffa-dd92eb2753e1

https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/38322563/bbd3c5f8-ceaa-46d6-ba65-62bb28988668

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    });
+    cy.wait(5000);
+  });
+
+  it('check UI Metric are being recorded', function () {
+    cy.intercept('api/ui_metric/report', cy.spy().as('reportreq'));
+
+    miscUtils.visitPage('app/home#/');
+
+    cy.wait(45000);
```
Comment (ashwin-pc) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js:31]: Lets not add wait statements to the test. Cant we wait for the loader? There should be a helper function to do this. You can do something similar for the other waits as well

Comment (kavilla) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js:31]: +1

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    cy.get('@reportreq').should('be.called');
+
+    // Now verify the response of api/stat
+
+    cy.wait(5000);
+
+    cy.request(
+      'GET',
+      'http://localhost:5601/api/stats?extended=true&legacy=true&exclude_usage=false'
```
Comment (kavilla) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js:41]: pull from cypress settings the dashboards url.

Comment (kavilla) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js:41]: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress.json#L5

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js (hunk 1 above)
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js]: Updated the tests to remove wait statements. Instead we are now waiting for the API call to be successful. @ashwin-pc @kavilla

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js (hunk 2 above)
Comment (LDrago27) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/telemetry/ui_metric.spec.js]: @kavilla Update to the use base path from config.

**Comment by kavilla** [comment]
Not necessarily a blocker but I would also add a new setting to the config: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress.json#L26

**Comment by LDrago27** [comment]
@kavilla that sounds good. I can update the tests so that they are being guarded by the config.

**Comment by kavilla** [comment]
Also, might need to validate the addition to a ciGroup here: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package.json#L31

And see if this test gets ran. The ciGroups are ran from the OSD github ci:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/cypress_workflow.yml#L122

**Comment by kavilla** [comment]
Note for after merging. If you would for these tests to then be executed from the build CI then we will need to update the test manifest:
https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.14.0/opensearch-dashboards-2.14.0-test.yml#L14

**Comment by kavilla** [comment]
Also for after merging updating OSD too if you would like it to be ran in our GitHub ci: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/cypress_workflow.yml#L37. But might have to be cautious about that. We have been hitting water marks for memory so depends on how much memory UI metric consumes.

**Pull request closed without merging**